core/templates/dev/head/pages/exploration_player/ExtractImageFilenamesFromStateService.js

```diff
@@ -108,30 +108,108 @@ oppia.factory('ExtractImageFilenamesFromStateService', [
     };
 
     /**
-     * Extracts the filenames from the filepath-value attribute of the
+     * Extracts the filepath object from the filepath-value attribute of the
      * oppia-noninteractive-image tags in the strHtml(given string).
-     * @param {string} strHtml - The string from which the filenames of images
-     *                           should be extracted.
+     * @param {string} strHtml - The string from which the object of
+     *                           filepath should be extracted.
      */
     var _extractFilepathValueFromOppiaNonInteractiveImageTag = function(
         strHtml) {
-      var filenames = [];
+      var fileInfo = [];
       var dummyElement = document.createElement('div');
       dummyElement.innerHTML = (
         HtmlEscaperService.escapedStrToUnescapedStr(strHtml));
 
       var imageTagList = dummyElement.getElementsByTagName(
         'oppia-noninteractive-image');
       for (i = 0; i < imageTagList.length; i++) {
-        filenames.push(imageTagList[i].getAttribute('filepath-with-value'));
+        var filepathObject = JSON.parse((imageTagList[i].getAttribute(
```

Comment:
Remove one pair of brackets, and move everything after `JSON.parse(` to next line.

Reply:
Done!